web3_proxy/src/frontend/authorization.rs

```diff
@@ -1187,8 +1234,43 @@ impl Web3ProxyApp {
                                 None
                             };
 
-                        let rpc_key_id =
-                            Some(rpc_key_model.id.try_into().expect("db ids are never 0"));
+                        // Get the user_tier
+                        let user_model = user::Entity::find_by_id(rpc_key_model.user_id)
+                            .one(db_replica.conn())
+                            .await?
+                            .context("user model was not found, but every rpc_key should have a user")?;
+
+                        let balance = balance::Entity::find()
+                            .filter(balance::Column::UserId.eq(user_model.id))
+                            .one(db_replica.conn())
+                            .await?
+                            .context("related balance was not found but every user should have a balance")?
+                            .available_balance;
+
+                        let mut user_tier_model =
+                            user_tier::Entity::find_by_id(user_model.user_tier_id)
+                                .one(db_replica.conn())
+                                .await?
+                                .context("related user tier not found, but every user should have a tier")?;
+
+                        // TODO: Do the logic here, as to how to treat the user, based on balance and initial check
+                        // Clear the cache (not the login!) in the stats if a tier-change happens (clear, but don't modify roles)
+                        if user_tier_model.title == "Premium" && balance < Decimal::from(10) {
+                            // Find the equivalent downgraded user tier, and modify limits from there
+                            if let Some(downgrade_user_tier) = user_tier_model.downgrade_tier_id {
+                                user_tier_model =
+                                    user_tier::Entity::find_by_id(downgrade_user_tier)
+                                        .one(db_replica.conn())
+                                        .await?
+                                        .context("finding the downgrade user tier for premium did not work for premium")?;
+                            } else {
+                                return Web3ProxyResult::Err(Web3ProxyError::InvalidUserTier);
```

Comment:
i don't think we want an error if this happens. they are just part of a tier that doesn't have downgrades enabled. There will definitely be accounts like that (unlimited accounts used for internal things like L2 syncing)

Reply:
this segment is within an if-else branch that assumes `if user_tier_model.title == "Premium" && balance < Decimal::from(10) {`. I think we do want it to error in this case